### ISSUE TITLE
Missing C library: LLVM-3.9.1

### ISSUE THREAD

**Issue by idontgetoutmuch** [bscarlet/llvm-general#207]
I installed llvm with shared libraries:

```
brew install llvm --with-shared-libs
```

But I now get

```
llvm-general/llvm-general $ git branch
git branch
  llvm-3.8
* llvm-3.9
  master
```

```
cabal configure -fshared-llvm --extra-lib-dirs=/usr/local/opt/llvm/lib --extra-include-dirs=/usr/local/opt/llvm/include
Warning: The package list for 'hackage.haskell.org' is 30 days old.
Run 'cabal update' to get the latest list of available packages.
Resolving dependencies...
Configuring llvm-general-3.9.0.0...
Warning: Instead of 'cc-options: -I/usr/local/Cellar/llvm/3.9.1/include' use
'include-dirs: /usr/local/Cellar/llvm/3.9.1/include'
setup: Missing dependency on a foreign library:
* Missing C library: LLVM-3.9.1
This problem can usually be solved by installing the system package that
provides this library (you may need the "-dev" version). If the library is
already installed but in a non-standard location then you can use the flags
--extra-include-dirs= and --extra-lib-dirs= to specify where it is.
```

**Comment by idontgetoutmuch** [comment]
Setting `cabal install --verbose=3` I get

```
/usr/local/bin/gcc-6 returned ExitFailure 1 with error message:
gcc-6: error: unrecognized command line option '-stdlib=libc++'
gcc-6: error: unrecognized command line option '-stdlib=libc++'
```

and

```
/usr/bin/gcc returned ExitFailure 1 with error message:
error: invalid argument '-std=c++11' not allowed with 'C/ObjC'
```

**Comment by idontgetoutmuch** [comment]
With

```
ignoredCxxFlags =
  [{- "-stdlib=libc++", -} "-std=c++11", "-Wcovered-switch-default", "-fcolor-diagnostics"] ++ map ("-D" ++) uncheckedHsFFIDefines
```

I seem to get further but now I get

```
/usr/bin/gcc returned ExitFailure 1 with error message:
ld: library not found for -lLLVM-3.9.1
clang: error: linker command failed with exit code 1 (use -v to see
invocation)
```

There are many dynamic libraries in `/usr/local/Cellar/llvm/3.9.1/lib` but `libLLVM-3.9.1` is not one of them.

```
ls /usr/local/Cellar/llvm/3.9.1/lib
BugpointPasses.dylib
LLVMHello.dylib
LLVMPolly.so
clang
cmake
libLLVMAArch64AsmParser.dylib
libLLVMAArch64AsmPrinter.dylib
libLLVMAArch64CodeGen.dylib
libLLVMAArch64Desc.dylib
libLLVMAArch64Disassembler.dylib
libLLVMAArch64Info.dylib
libLLVMAArch64Utils.dylib
libLLVMAMDGPUAsmParser.dylib
libLLVMAMDGPUAsmPrinter.dylib
libLLVMAMDGPUCodeGen.dylib
libLLVMAMDGPUDesc.dylib
libLLVMAMDGPUDisassembler.dylib
libLLVMAMDGPUInfo.dylib
libLLVMAMDGPUUtils.dylib
libLLVMARMAsmParser.dylib
libLLVMARMAsmPrinter.dylib
libLLVMARMCodeGen.dylib
libLLVMARMDesc.dylib
libLLVMARMDisassembler.dylib
libLLVMARMInfo.dylib
libLLVMAnalysis.dylib
libLLVMAsmParser.dylib
libLLVMAsmPrinter.dylib
libLLVMBPFAsmPrinter.dylib
libLLVMBPFCodeGen.dylib
libLLVMBPFDesc.dylib
libLLVMBPFInfo.dylib
libLLVMBitReader.dylib
libLLVMBitWriter.dylib
libLLVMCodeGen.dylib
libLLVMCore.dylib
libLLVMCoverage.dylib
libLLVMDebugInfoCodeView.dylib
libLLVMDebugInfoDWARF.dylib
libLLVMDebugInfoPDB.dylib
libLLVMExecutionEngine.dylib
libLLVMGlobalISel.dylib
libLLVMHexagonAsmParser.dylib
libLLVMHexagonCodeGen.dylib
libLLVMHexagonDesc.dylib
libLLVMHexagonDisassembler.dylib
libLLVMHexagonInfo.dylib
libLLVMIRReader.dylib
libLLVMInstCombine.dylib
libLLVMInstrumentation.dylib
libLLVMInterpreter.dylib
libLLVMLTO.dylib
libLLVMLibDriver.dylib
libLLVMLineEditor.dylib
libLLVMLinker.dylib
libLLVMMC.dylib
libLLVMMCDisassembler.dylib
libLLVMMCJIT.dylib
libLLVMMCParser.dylib
libLLVMMIRParser.dylib
libLLVMMSP430AsmPrinter.dylib
libLLVMMSP430CodeGen.dylib
libLLVMMSP430Desc.dylib
libLLVMMSP430Info.dylib
libLLVMMipsAsmParser.dylib
libLLVMMipsAsmPrinter.dylib
libLLVMMipsCodeGen.dylib
libLLVMMipsDesc.dylib
libLLVMMipsDisassembler.dylib
libLLVMMipsInfo.dylib
libLLVMNVPTXAsmPrinter.dylib
libLLVMNVPTXCodeGen.dylib
libLLVMNVPTXDesc.dylib
libLLVMNVPTXInfo.dylib
libLLVMObjCARCOpts.dylib
libLLVMObject.dylib
libLLVMObjectYAML.dylib
libLLVMOption.dylib
libLLVMOrcJIT.dylib
libLLVMPasses.dylib
libLLVMPowerPCAsmParser.dylib
libLLVMPowerPCAsmPrinter.dylib
libLLVMPowerPCCodeGen.dylib
libLLVMPowerPCDesc.dylib
libLLVMPowerPCDisassembler.dylib
libLLVMPowerPCInfo.dylib
libLLVMProfileData.dylib
libLLVMRuntimeDyld.dylib
libLLVMScalarOpts.dylib
libLLVMSelectionDAG.dylib
libLLVMSparcAsmParser.dylib
libLLVMSparcAsmPrinter.dylib
libLLVMSparcCodeGen.dylib
libLLVMSparcDesc.dylib
libLLVMSparcDisassembler.dylib
libLLVMSparcInfo.dylib
libLLVMSupport.dylib
libLLVMSymbolize.dylib
libLLVMSystemZAsmParser.dylib
libLLVMSystemZAsmPrinter.dylib
libLLVMSystemZCodeGen.dylib
libLLVMSystemZDesc.dylib
libLLVMSystemZDisassembler.dylib
libLLVMSystemZInfo.dylib
libLLVMTableGen.dylib
libLLVMTarget.dylib
libLLVMTransformUtils.dylib
libLLVMVectorize.dylib
libLLVMX86AsmParser.dylib
libLLVMX86AsmPrinter.dylib
libLLVMX86CodeGen.dylib
libLLVMX86Desc.dylib
libLLVMX86Disassembler.dylib
libLLVMX86Info.dylib
libLLVMX86Utils.dylib
libLLVMXCoreAsmPrinter.dylib
libLLVMXCoreCodeGen.dylib
libLLVMXCoreDesc.dylib
libLLVMXCoreDisassembler.dylib
libLLVMXCoreInfo.dylib
libLLVMipo.dylib
libLTO.dylib
libPolly.dylib
libPollyISL.a
libPollyPPCG.a
libc++.1.0.dylib
libc++.1.dylib
libc++.dylib
libclang.dylib
libclangARCMigrate.dylib
libclangAST.dylib
libclangASTMatchers.dylib
libclangAnalysis.dylib
libclangApplyReplacements.dylib
libclangBasic.dylib
libclangCodeGen.dylib
libclangDriver.dylib
libclangDynamicASTMatchers.dylib
libclangEdit.dylib
libclangFormat.dylib
libclangFrontend.dylib
libclangFrontendTool.dylib
libclangIncludeFixer.dylib
libclangIndex.dylib
libclangLex.dylib
libclangParse.dylib
libclangQuery.dylib
libclangRename.dylib
libclangRewrite.dylib
libclangRewriteFrontend.dylib
libclangSema.dylib
libclangSerialization.dylib
libclangStaticAnalyzerCheckers.dylib
libclangStaticAnalyzerCore.dylib
libclangStaticAnalyzerFrontend.dylib
libclangTidy.dylib
libclangTidyBoostModule.dylib
libclangTidyCERTModule.dylib
libclangTidyCppCoreGuidelinesModule.dylib
libclangTidyGoogleModule.dylib
libclangTidyLLVMModule.dylib
libclangTidyMiscModule.dylib
libclangTidyModernizeModule.dylib
libclangTidyPerformanceModule.dylib
libclangTidyPlugin.dylib
libclangTidyReadabilityModule.dylib
libclangTidyUtils.dylib
libclangTooling.dylib
libclangToolingCore.dylib
libfindAllSymbols.dylib
libgomp.dylib
libiomp5.dylib
liblldCOFF.dylib
liblldConfig.dylib
liblldCore.dylib
liblldDriver.dylib
liblldELF.dylib
liblldMachO.dylib
liblldReaderWriter.dylib
liblldYAML.dylib
libomp.dylib
libunwind.1.0.dylib
libunwind.1.dylib
libunwind.dylib
python2.7
```

**Comment by idontgetoutmuch** [comment]
I reinstalled llvm with `brew install llvm` and setting

```
ls -l /usr/local/Cellar/llvm/3.9.1/lib/libLLVM-3.9.1.dylib 
lrwxr-xr-x  1 dom  admin  46 Mar  4 06:52 /usr/local/Cellar/llvm/3.9.1/lib/libLLVM-3.9.1.dylib -> /usr/local/Cellar/llvm/3.9.1/lib/libLLVM.dylib
```

`cabal configure -fshared-llvm` succeeds but sadly I now get

```
dyld: Library not loaded: @rpath/libLTO.dylib
  Referenced from: /usr/local/opt/llvm/lib/libLLVM.dylib
  Reason: image not found
running dist/build/LLVM/General/Internal/FFI/LLVMCTypes_hsc_make failed (exit code -5)
command was: dist/build/LLVM/General/Internal/FFI/LLVMCTypes_hsc_make  >dist/build/LLVM/General/Internal/FFI/LLVMCTypes.hs
```

I can see `libLTO.dylib` in `/usr/local/Cellar/llvm/3.9.1/lib` :(

**Comment by idontgetoutmuch** [comment]
```
export DYLD_FALLBACK_LIBRARY_PATH=/usr/local/Cellar/llvm/3.9.1/lib
export DYLD_FALLBACK_LIBRARY_PATH=/usr/local/Cellar/llvm/3.9.1/lib
cabal build
cabal build
Building llvm-general-3.9.0.0...
Preprocessing library llvm-general-3.9.0.0...

dist/build/LLVM/General/Internal/LibraryFunction.hs:1:1: error:
    File name does not match module name:
    Saw: ‘Main’
    Expected: ‘LLVM.General.Internal.LibraryFunction’
```

**Comment by idontgetoutmuch** [comment]
I am not sure that exporting solved that problem with `libLTO.dylib`

```
/llvm-general/llvm-general $ cabal configure -fshared-llvm
cabal configure -fshared-llvm
Warning: The package list for 'hackage.haskell.org' is 30 days old.
Run 'cabal update' to get the latest list of available packages.
Resolving dependencies...
[1 of 1] Compiling Main             ( dist/setup/setup.hs, dist/setup/Main.o )
Linking ./dist/setup/setup ...
Configuring llvm-general-3.9.0.0...
Warning: Instead of 'cc-options: -I/usr/local/Cellar/llvm/3.9.1/include' use
'include-dirs: /usr/local/Cellar/llvm/3.9.1/include'
~/Dropbox/Private/NumMethHaskell/llvm-general/llvm-general $ cabal build
cabal build
Building llvm-general-3.9.0.0...
Preprocessing library llvm-general-3.9.0.0...
In file included from dist/build/LLVM/General/Internal/LibraryFunction_hsc_make.c:1:
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:34:20: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_const(x...)                               \
                   ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:40:24: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_const_str(x...)                                       \
                       ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:59:19: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_type(t...)                                      \
                  ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:85:19: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_size(t...) \
                  ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:88:24: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_alignment(t...) \
                       ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:101:25: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_haskellize(x...)                                       \
                        ^
6 warnings generated.
In file included from dist/build/LLVM/General/Internal/LibraryFunction_hsc_utils.c:7:
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:34:20: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_const(x...)                               \
                   ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:40:24: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_const_str(x...)                                       \
                       ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:59:19: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_type(t...)                                      \
                  ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:85:19: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_size(t...) \
                  ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:88:24: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_alignment(t...) \
                       ^
/Library/Frameworks/GHC.framework/Versions/8.0.1-x86_64/usr/lib/ghc-8.0.1/template-hsc.h:101:25: warning: 
      named variadic macros are a GNU extension [-Wvariadic-macros]
#define hsc_haskellize(x...)                                       \
                        ^
6 warnings generated.
dyld: Library not loaded: @rpath/libLTO.dylib
  Referenced from: /usr/local/opt/llvm/lib/libLLVM.dylib
  Reason: image not found
running dist/build/LLVM/General/Internal/LibraryFunction_hsc_make failed (exit code -5)
command was: dist/build/LLVM/General/Internal/LibraryFunction_hsc_make  >dist/build/LLVM/General/Internal/LibraryFunction.hs
```

**Comment by idontgetoutmuch** [comment]
Using @borsboom's workaround (https://github.com/commercialhaskell/stack/issues/1161#issuecomment-186690904) I now get compilation which then errors :(

```
[101 of 101] Compiling LLVM.General     ( src/LLVM/General.hs, dist/build/LLVM/General.o )

In file included from src/LLVM/General/Internal/FFI/AssemblyC.cpp:2:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/AsmParser/Parser.h:17:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/Support/MemoryBuffer.h:17:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/Twine.h:13:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:17:0: error:
    

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/iterator_range.h:39:28: error:
     warning: rvalue references are a C++11 extension [-Wc++11-extensions]
      iterator_range(Container &&c)
                               ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/iterator_range.h:63:65: error:
     warning: rvalue references are a C++11 extension [-Wc++11-extensions]
iterator_range<decltype(begin(std::declval<T>()))> drop_begin(T &&t, int n) {
                                                                ^

In file included from src/LLVM/General/Internal/FFI/AssemblyC.cpp:2:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/AsmParser/Parser.h:17:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/Support/MemoryBuffer.h:17:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/Twine.h:13:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:18:0: error:
    

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:41:25: error:
     warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
      AlignmentCalcImpl() = delete;
                            ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:58:24: error:
     warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
      ~AlignmentCalcImpl() override = 0;
                           ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:78:10: error:
     error: unknown type name 'constexpr'
      static constexpr unsigned Alignment = static_cast<unsigned int>(
             ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:78:20: error:
     error: expected member name or ';' after declaration specifiers
      static constexpr unsigned Alignment = static_cast<unsigned int>(
      ~~~~~~~~~~~~~~~~ ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:86:42: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_GreaterEqual_2Bytes = Alignment >= 2 ? 1 : 0 };
                                             ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:87:42: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_GreaterEqual_4Bytes = Alignment >= 4 ? 1 : 0 };
                                             ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:88:42: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_GreaterEqual_8Bytes = Alignment >= 8 ? 1 : 0 };
                                             ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:89:43: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_GreaterEqual_16Bytes = Alignment >= 16 ? 1 : 0 };
                                              ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:91:39: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_LessEqual_2Bytes = Alignment <= 2 ? 1 : 0 };
                                          ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:92:39: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_LessEqual_4Bytes = Alignment <= 4 ? 1 : 0 };
                                          ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:93:39: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_LessEqual_8Bytes = Alignment <= 8 ? 1 : 0 };
                                          ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:94:40: error:
     error: use of undeclared identifier 'Alignment'
      enum { Alignment_LessEqual_16Bytes = Alignment <= 16 ? 1 : 0 };
                                           ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:98:23: error:
     error: unknown type name 'constexpr'
template <typename T> constexpr unsigned AlignOf<T>::Alignment;
                      ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:98:33: error:
     error: expected unqualified-id
template <typename T> constexpr unsigned AlignOf<T>::Alignment;
                                ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/Support/AlignOf.h:226:19: error:
     warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
      AlignerImpl() = delete;
                      ^

In file included from src/LLVM/General/Internal/FFI/AssemblyC.cpp:2:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/AsmParser/Parser.h:17:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/Support/MemoryBuffer.h:17:0: error:
    

In file included from /usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/Twine.h:13:0: error:
    

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:215:20: error:
     warning: rvalue references are a C++11 extension [-Wc++11-extensions]
      void push_back(T &&Elt) {
                       ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:321:45: error:
     warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
      SmallVectorImpl(const SmallVectorImpl&) = delete;
                                                ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:411:20: error:
     error: no type named 'initializer_list' in namespace 'std'
      void append(std::initializer_list<T> IL) {
                  ~~~~~^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:411:36: error:
     error: expected ')'
      void append(std::initializer_list<T> IL) {
                                       ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:411:14: error:
     note: to match this '('
      void append(std::initializer_list<T> IL) {
                 ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:423:20: error:
     error: no type named 'initializer_list' in namespace 'std'
      void assign(std::initializer_list<T> IL) {
                  ~~~~~^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:423:36: error:
     error: expected ')'
      void assign(std::initializer_list<T> IL) {
                                       ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:423:14: error:
     note: to match this '('
      void assign(std::initializer_list<T> IL) {
                 ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:461:33: error:
     warning: rvalue references are a C++11 extension [-Wc++11-extensions]
      iterator insert(iterator I, T &&Elt) {
                                    ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:628:32: error:
     error: no type named 'initializer_list' in namespace 'std'
      void insert(iterator I, std::initializer_list<T> IL) {
                              ~~~~~^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:628:48: error:
     error: expected ')'
      void insert(iterator I, std::initializer_list<T> IL) {
                                                   ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:628:14: error:
     note: to match this '('
      void insert(iterator I, std::initializer_list<T> IL) {
                 ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:632:21: error:
     warning: variadic templates are a C++11 extension [-Wc++11-extensions]
      template <typename... ArgTypes> void emplace_back(ArgTypes &&... Args) {
                        ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:632:62: error:
     warning: rvalue references are a C++11 extension [-Wc++11-extensions]
      template <typename... ArgTypes> void emplace_back(ArgTypes &&... Args) {
                                                                 ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:641:46: error:
     warning: rvalue references are a C++11 extension [-Wc++11-extensions]
      SmallVectorImpl &operator=(SmallVectorImpl &&RHS);
                                                 ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:356:12: error:
     warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]
          for (auto I = this->end(), E = this->begin() + N; I != E; ++I)
               ^

/usr/local/Cellar/llvm/3.9.1/include/llvm/ADT/SmallVector.h:412:12: error:
     error: use of undeclared identifier 'IL'
        append(IL.begin(), IL.end());
               ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
12 warnings and 20 errors generated.
`gcc' failed in phase `C Compiler'. (Exit code: 1)
```

**Comment by idontgetoutmuch** [comment]
Switching to `gcc-6` and now it installs!
